src/Features/Core/Portable/AddImport/SymbolReferenceFinder.cs

```diff
@@ -225,11 +239,25 @@ private ImmutableArray<SymbolReference> DeDupeAndSortReferences(ImmutableArray<S
                 ITypeSymbol symbol,
                 int arity,
                 bool inAttributeContext,
-                bool hasIncompleteParentMember)
+                bool hasIncompleteParentMember,
+                bool looksGeneric)
             {
-                return (arity == 0 || symbol.GetArity() == arity || hasIncompleteParentMember)
-                       && symbol.IsAccessibleWithin(_semanticModel.Compilation.Assembly)
-                       && (!inAttributeContext || symbol.IsAttribute());
+                if (inAttributeContext && !symbol.IsAttribute())
+                {
+                    return false;
+                }
+
+                if (!symbol.IsAccessibleWithin(_semanticModel.Compilation.Assembly))
```

Comment:
❓ Can this check be slow? If so and one of the other checks is likely to fail-fast early, we could move it later in this method.